lib/pages/classes/view/class_events_card.dart

```diff
@@ -0,0 +1,76 @@
+import 'package:acs_upb_mobile/pages/timetable/model/events/recurring_event.dart';
+import 'package:acs_upb_mobile/pages/timetable/model/events/uni_event.dart';
+import 'package:acs_upb_mobile/pages/timetable/service/uni_event_provider.dart';
+import 'package:acs_upb_mobile/pages/timetable/view/events/event_view.dart';
+import 'package:acs_upb_mobile/resources/locale_provider.dart';
+import 'package:acs_upb_mobile/widgets/info_card.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:acs_upb_mobile/generated/l10n.dart';
+
+class ClassEventsCard extends StatefulWidget {
+  const ClassEventsCard(this.currentClassId, {Key key}) : super(key: key);
+  final String currentClassId;
+
+  @override
+  _ClassEventsCardState createState() => _ClassEventsCardState();
+}
+
+class _ClassEventsCardState extends State<ClassEventsCard> {
+  @override
+  Widget build(BuildContext context) {
+    final UniEventProvider eventProvider =
+        Provider.of<UniEventProvider>(context);
+
+    return InfoCard<Iterable<UniEvent>>(
+      title: S.of(context).sectionEvents,
+      edgeInsets: EdgeInsets.zero,
+      future: eventProvider.getAllEventsOfClass(widget.currentClassId),
+      builder: (events) => Column(
+        children: events
+            .map(
+              (event) => ListTile(
+                key: ValueKey(event.id),
+                contentPadding: const EdgeInsets.symmetric(vertical: 8),
```

Comment:
Just built the code and I think this padding is unnecessary, it looks better without it.